protocol/v2/blockchain/beacon/network.go

```diff
@@ -11,17 +11,38 @@ import (
 // Network is a beacon chain network.
 type Network struct {
 	spectypes.BeaconNetwork
+	LocalTestNet bool
 }
 
 // NewNetwork creates a new beacon chain network.
 func NewNetwork(network spectypes.BeaconNetwork) Network {
-	return Network{network}
+	return Network{
+		BeaconNetwork: network,
+		LocalTestNet:  false,
+	}
+}
+
+// NewNetworkWithLocalTestNet creates a new beacon chain network.
+func NewNetworkWithLocalTestNet(network spectypes.BeaconNetwork, localTestNet bool) Network {
+	return Network{
+		BeaconNetwork: network,
+		LocalTestNet:  localTestNet,
+	}
+}
+
+// CustomMinGenesisTime returns min genesis time value
+func (n Network) CustomMinGenesisTime() uint64 {
```

Comment:
I think setting custom value should remain an abstraction, so perhaps just `MinGenesisTime`? It would override `spectypes.BeaconNetwork`'s `MinGenesisTime`

Reply:
This can actually be pretty good and help us have less changes.